xcode/Subconscious/Shared/Components/Common/Byline/PetnameView.swift

```diff
@@ -7,54 +7,111 @@
 
 import SwiftUI
 
+public enum NameVariant {
+    case known(Slashlink, Petname.Name)
+    case named(Slashlink, Petname.Name)
+    case unknown(Slashlink)
```

Comment:
Might consider type names that map directly to concepts from one of the petname papers? https://spritely.institute/static/papers/petnames.html

- `petname` (following directly)
- `selfNickname` or `autonym` or `selfProposedName` (the nickname a sphere proposes for itself in its header)
- `nickname` or `proposedName` (The nickname a 3p proposes for another sphere. This is the petname that sphere gives another sphere.)

Reply:
Yes, this is better. I dropped the `unknown` case reworked the logic, introducing `UserProfile.toNameVariant()` which can hold all the logic about this mapping.